tensorboard/compat/tensorflow_stub/io/gfile_fsspec_test.py

```diff
@@ -0,0 +1,443 @@
+                    "some_flume_output.txt",
+                    "some_more_flume_output.txt",
+                ],
+            ],
+            [
+                "quuz",
+                [
+                    "e.tfevents.1",
+                ],
+            ],
+            [
+                "quuz/garply",
+                [
+                    "f.tfevents.1",
+                ],
+            ],
+            [
+                "quuz/garply/corge",
+                [
+                    "g.tfevents.1",
+                ],
+            ],
+            [
+                "quuz/garply/grault",
+                [
+                    "h.tfevents.1",
+                ],
+            ],
+            ["waldo", []],
+            [
+                "waldo/fred",
+                [
+                    "i.tfevents.1",
+                ],
+            ],
+        ]
+        for pair in expected:
+            # If this is not the top-level directory, prepend the high-level
+            # directory.
+            pair[0] = (
+                os.path.join(temp_dir, pair[0].replace("/", os.path.sep))
+                if pair[0]
+                else temp_dir
+            )
+        gotten = gfile.walk(temp_dir)
+        self._CompareFilesPerSubdirectory(expected, gotten)
+
+    def testStat(self):
+        temp_dir = self.get_temp_dir()
+        self._CreateDeepDirectoryStructure(temp_dir)
+        ckpt_path = os.path.join(temp_dir, "model.ckpt")
+        ckpt_content = "asdfasdfasdffoobarbuzz"
+        with fsspec.open(ckpt_path, "w") as f:
+            f.write(ckpt_content)
+        ckpt_stat = gfile.stat(ckpt_path)
+        self.assertEqual(ckpt_stat.length, len(ckpt_content))
+        bad_ckpt_path = os.path.join(temp_dir, "bad_model.ckpt")
+        with self.assertRaises(errors.NotFoundError):
+            gfile.stat(bad_ckpt_path)
+
+    def testRead(self):
+        temp_dir = self.get_temp_dir()
+        self._CreateDeepDirectoryStructure(temp_dir)
+        ckpt_path = os.path.join(temp_dir, "model.ckpt")
+        ckpt_content = "asdfasdfasdffoobarbuzz"
+        with fsspec.open(ckpt_path, "w") as f:
+            f.write(ckpt_content)
+        with gfile.GFile(ckpt_path, "r") as f:
+            f.buff_chunk_size = 4  # Test buffering by reducing chunk size
+            ckpt_read = f.read()
+            self.assertEqual(ckpt_content, ckpt_read)
+
+    def testTextMode(self):
+        temp_dir = self.get_temp_dir()
+        self._CreateDeepDirectoryStructure(temp_dir)
+        ckpt_path = os.path.join(temp_dir, "model.ckpt")
+
+        # Write out newlines as given (i.e., \r\n) regardless of OS, so as to
+        # test translation on read.
+        with fsspec.open(ckpt_path, "w", newline="") as f:
+            data = "asdf\nasdf\nasdf\n"
+            f.write(data)
+        with gfile.GFile(ckpt_path, "r") as f:
+            f.buff_chunk_size = 4  # Test buffering by reducing chunk size
+            with self.assertRaises(errors.InvalidArgumentError):
```

Comment:
Why is it expected that this would raise InvalidArgumentError?

Reply:
the fsspec implementation here doesn't support seekable streams for text, though digging deeper into it I'm actually not quite sure why here. It uses io.TextIOWrapper which does support them, need to take a closer look at this

Reply:
https://github.com/intake/filesystem_spec/pull/743 seems to be a bug with fsspec local filesystem in text mode.

Reply:
I just removed that check for now. We can add it once that's fixed. Don't want to add a check for the bad behavior only for it to fail on dep update.